### PR TITLE
Refs #31233 -- Added DatabaseWrapper._maindb_cursor() on Oracle.

### PR DESCRIPTION
**Pull request by felixxm**
Follow up to f48f671223a20b161ca819cf7d6298e43b8ba5fe.

### CODE REVIEW

File: django/db/backends/oracle/creation.py [hunk 1]
```diff
@@ -13,9 +14,8 @@ class DatabaseCreation(BaseDatabaseCreation):
     @cached_property
     def _maindb_connection(self):
         """
-        This is analogous to other backends' `_nodb_connection` property,
-        which allows access to an "administrative" connection which can
-        be used to manage the test databases.
+        Allow access to an "administrative" connection which can be used to
```
Comment (felixxm) [inline django/db/backends/oracle/creation.py:17]: `_nodb_connection` was removed in f48f671223a20b161ca819cf7d6298e43b8ba5fe.

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Comment by akshaykedar1996** [comment]
In Django, the DatabaseWrapper class is used to manage connections to a database. The _cursor() method of DatabaseWrapper is used to create a cursor object for executing SQL queries on the database.

Prior to the change made in ticket #16583, there was no built-in support for creating a cursor that operates exclusively on the main database in Oracle. Instead, developers had to manually specify the name of the main database when creating a cursor.

In ticket #31233, a new method _maindb_cursor() was added to DatabaseWrapper for Oracle databases. This method creates a cursor that operates exclusively on the main database without requiring the developer to manually specify the database name.

To create a cursor that operates exclusively on the main database in Oracle, developers can now use the following code:

python

from django.db import connections

cursor = connections['my_oracle_db'].cursor()
main_cursor = connections['my_oracle_db']._maindb_cursor()

In the example above, cursor() is used to create a cursor that operates on the entire Oracle database. _maindb_cursor() is used to create a cursor that operates exclusively on the main database.

Overall, the addition of _maindb_cursor() for Oracle databases provides a more convenient and efficient way to create cursors that operate on the main database, which is an important feature for many Django applications that use Oracle as a backend database.

**Review by David-Wobrock — APPROVED** [review]

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Comment by felixxm** [comment]
Unfortunately, this introduces a huge performance regression in creating test databases.

**Pull request closed without merging**